src/snowcli/output/printing.py

```diff
@@ -23,6 +23,75 @@ def default(self, o):
         return super().default(o)
 
 
+class OutputData:
+    def __init__(
+        self,
+        format: Optional[OutputFormat] = None,
+        cursor: Optional[SnowflakeCursor] = None,
+        string_output: Optional[str] = None,
+        data: Optional[List[Dict]] = None,
+    ) -> None:
+        self.output: List[Union[SnowflakeCursor, str, List[Dict]]] = []
+        self.counter = 0
+
+        if format:
+            self.format = format
+        else:
+            self.format = _get_format_type()
+
+        if cursor:
+            self.add_cursor(cursor)
+
+        if string_output:
+            self.add_string_output(string_output)
+
+        if data:
+            self.add_data(data)
```

Comment:
Instead of that I would consider builder pattern.
```
output = OutputData().add_string("ddd").add_cursor(curr).add_data([1,2,3])
```
what do you think?

Reply:
I like it, changed